admin/data/sql/migrations/20200129155749.do.fix-db-routine.sql

```diff
@@ -0,0 +1,113 @@
+CREATE OR ALTER FUNCTION dbo.ufnCalcCheckStatusID (@checkId int)
+    RETURNS int
+AS
+BEGIN
+    DECLARE
+        @checkTimeLimit tinyint,
+        @complete bit,
+        @isLiveCheck bit,
+        @pinExpiresAt datetimeoffset,
+        @processingFailed bit,
+        @pupilLoginDate datetimeoffset,
+        @receivedByServerAt datetimeoffset,
+        @ret int
+    ;
+
+    SELECT
+        TOP 1
+            @checkTimeLimit = checkTimeLimit
+      FROM
+          [mtc_admin].[settings];
+
+    SELECT
+            @complete = complete,
+            @isLiveCheck = isLiveCheck,
+            @ret = checkStatus_id, -- set the return val to the current status
+            @pinExpiresAt = pinExpiresAt,
+            @processingFailed = processingFailed,
+            @pupilLoginDate = pupilLoginDate,
+            @receivedByServerAt = receivedByServerAt
+      FROM
+          [mtc_admin].[check] c LEFT OUTER JOIN
+          [mtc_admin].[checkPin] cp ON (c.id = cp.check_id)
+     WHERE
+             c.id = @checkId;
+
+    -- NEW
+    IF (@pupilLoginDate IS NULL AND
+        @receivedByServerAt IS NULL AND
+        @pinExpiresAt IS NOT NULL AND
+        @pinExpiresAt > GETUTCDATE())
+        BEGIN
+            SELECT @ret = id FROM [mtc_admin].[checkStatus] where code = 'NEW';
+            RETURN @ret;
+        END
+
+    -- EXPIRED
+    IF ((@pupilLoginDate IS NULL OR
+         @pupilLoginDate IS NOT NULL AND @isLiveCheck = 0) AND
```

Comment:
is this correct `@isLiveCheck = 0`?  that would only point to try out expired checks?

Reply:
Live checks only go to expired if no-one logged in.
TIO checks go to expired even if they were used.